src/yunohost/app.py

```diff
@@ -1659,6 +1721,129 @@ def _parse_app_instance_name(app_instance_name):
     app_instance_nb = int(match.groupdict().get('appinstancenb')) if match.groupdict().get('appinstancenb') is not None else 1
     return (appid, app_instance_nb)
 
+
+def _using_legacy_applist_system():
+    """
+    Return True if we're using the old fetchlist scheme.
+    This is determined by the presence of some cron job yunohost-applist-foo
+    """
+
+    return (glob.glob("/etc/cron.d/yunohost-applist-*") != [])
+
+
+def _migrate_applist_system():
+    """
+    Migrate from the legacy fetchlist system to the new one
+    """
+    legacy_crons = glob.glob("/etc/cron.d/yunohost-applist-*")
+
+    for cron_path in legacy_crons:
+        applist_name = os.path.basename(cron_path).replace("yunohost-applist-", "")
+        logger.info(m18n.n('appslist_migrating', name=applist_name))
+
+        # Parse applist url in cron
+        cron_file_content = open(cron_path).read().strip()
+        applist_url_parse = re.search("-u (https?://[^ ]+)", cron_file_content)
+
+        # Abort if we did not find an url
+        if not applist_url_parse or not applist_url_parse.groups():
+            # Bkp the old cron job somewhere else
+            bkp_file = "/etc/yunohost/%s.oldlist.bkp" % applist_name
+            os.rename(cron_path, bkp_file)
+            # Notice the user
+            logger.warning(m18n.n('appslist_could_not_migrate',
+                           name=applist_name,
+                           bkp_file=bkp_file))
+        # Otherwise, register the list and remove the legacy cron
+        else:
+            applist_url = applist_url_parse.groups()[0]
+            try:
+                _register_new_applist(applist_url, applist_name)
+            # Might get an exception if two legacy cron jobs conflict
+            # in terms of url...
+            except Exception as e:
+                logger.error(str(e))
+                # Bkp the old cron job somewhere else
+                bkp_file = "/etc/yunohost/%s.oldlist.bkp" % applist_name
+                os.rename(cron_path, bkp_file)
+                # Notice the user
+                logger.warning(m18n.n('appslist_could_not_migrate',
+                               name=applist_name,
+                               bkp_file=bkp_file))
+            else:
+                os.remove(cron_path)
+
+    _install_applist_fetch_cron()
+
+
+def _install_applist_fetch_cron():
+
+    logger.debug("Installing applist fetch cron job")
+
+    with open("/etc/cron.daily/yunohost-fetch-applists", "w") as f:
+        f.write('#!/bin/bash\nyunohost app fetchlist > /dev/null 2>&1\n')
+
+
+def _read_applist_list():
+    """
+    Read the json corresponding to the list of applists
+    """
+
+    # If file does not exists yet, return empty dict
+    if not os.path.exists(APPLISTS_JSON):
+        return {}
+
+    # Read file content
+    with open(APPLISTS_JSON, "r") as f:
```

Comment:
I started a related PR here : https://github.com/YunoHost/yunohost/pull/274 
This is going to take quite some time so let's consider these PR independently, then we'll move to the common io helpers once they're here.